Renamer/Culture.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System.Collections.Generic;
+using System.Collections;
+using System;
+using System.IO;
+using System.Reflection;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Events;
+using System.Linq;
+
+namespace Renamer
+{
+    public class Culture
+    {
+        public bool femaleSurnamesExist = false;
+        public string cultureName = "";
+        public string[] fnames1 = { };
+        public string[] fnames2 = { };
+        public string[] fnames3 = { };
+        public string[] mnames1 = { };
+        public string[] mnames2 = { };
+        public string[] mnames3 = { };
+        public string[] lnames1 = { };
+        public string[] lnames2 = { };
+        public string[] lnames3 = { };
+        public string[] flnames1 = { };
+        public string[] flnames2 = { };
+        public string[] flnames3 = { };
+
+        public Culture(ConfigNode node)
+        {
+            ConfigNode temp;
+            string[] vals;
+            if (node.HasValue("name"))
+            {
+                cultureName = node.GetValue("name");
+            }
+            if (node.HasNode("FFIRSTNAME1"))
```

Comment:
Looks like you're doing a lot of similar code here.  I'd suggest creating a list of nodenames that you care about, and then iterating on that list, returning the values for that name.  Then perhaps you could use a dictionary to map the node name with the valid values for that name, that way you don't need all those separate string arrays

Reply:
I've cleaned this up a bit.  I implemented a case/switch instead.  It's still a bit long but it doesn't repeat much code.